src/share/classes/com/sun/javatest/regtest/exec/Agent.java

```diff
@@ -212,6 +229,46 @@ private Agent(File dir, JDK jdk, List<String> vmOpts, Map<String, String> envVar
         }
     }
 
+    // accept()s a connection over the ServerSocket and then reads handshake bytes
+    // from the connected socket. if the read bytes match the expected handshake bytes
+    // then the connection and handshake is considered successful and the accepted
+    // Socket is returned. if the read bytes don't match the expected handshake bytes
+    // or if the read times out after accept()ing an connection, then this method
+    // closes the accepted connection and returns null.
+    private Socket acceptAndHandshake(final ServerSocket ss, final int handshakeReadTimeout)
+            throws IOException {
+        final Socket s = ss.accept();
+        try {
+            log("Received connection on port " + ss.getLocalPort() + " from " + s);
+            s.setSoTimeout(handshakeReadTimeout);
+            // read the handshake bytes
+            final byte[] handshakeBytes = new byte[JTREG_AGENT_HANDSHAKE_MAGIC.length];
+            final int numRead = s.getInputStream().read(handshakeBytes);
+            if (numRead == -1) {
+                // EOF
+                log("Received EOF before handshake from " + s);
+                return null;
+            }
+            // verify the handshake bytes
+            if (!Arrays.equals(JTREG_AGENT_HANDSHAKE_MAGIC, handshakeBytes)) {
+                log("Unexpected handshake bytes from socket: " + s + ", expected: "
+                        + Arrays.toString(JTREG_AGENT_HANDSHAKE_MAGIC)
+                        + ", actual: " + Arrays.toString(handshakeBytes));
+                return null;
+            }
+        } catch (IOException ioe) {
+            log("closing connection to " + s + " due to: " + ioe);
+            try {
+                s.close();
+            } catch (IOException ignored) {
+                // ignore
```

Comment:
Please don't swallow exceptions.

Reply:
Use `Throwable.addSuppressed` -- it is for exactly this situation.

Reply:
> Use Throwable.addSuppressed -- it is for exactly this situation.

But we wouldn't be throwing any exception to which we would be adding this as a suppressed exception. This failure is during close()ing a connection on which we either had a read timeout or an IOException when reading off it. We log that failure a few lines above this and then go and close this connection and then attempt to accept() a next connection (if any).

I can update this catch block to log a message which states the close() failed, along with the reason. Is that reasonable?

Reply:
I've now updated the PR to log the exception stacktrace if/when it fails. This should help debug any unexpected failures in this part of the code.